### PR TITLE
feat: adding file rename functionality and tests

### PR DESCRIPTION
**Pull request by demanr**
## Resolving Issues
#233 #236

## Description
This pull request implements file rename support in the text replay scrubber component. The changes include:

- Added `ReplayEvent` type definitions for `DID_CHANGE_TEXT_DOCUMENT` and `DID_RENAME_FILES` events
- Implemented `getVisibleFilePathsFromTimeline()` function to track file paths that are visible at any point in the timeline, handling renames by removing old paths and adding new ones
- Added `replayFileContents()` function to replay file content changes while preserving content across renames
- Updated the scrubber component to track currently visible files based on replay position rather than all files in the transcript
- Enhanced file selection logic to handle renamed files by automatically switching to the new file path when the selected file is renamed
- Modified content replay to maintain separate file contents for all visible files and handle file renames by copying content to the new path
- Added comprehensive test coverage for rename functionality, including edge cases like reusing old file names after renames

The file tabs now correctly show only files that exist at the current replay position, and the selected file automatically follows renames during playback.

How to Test
Create a replay as usual (ensuring you update the extension name to your workspace), while changing some file names and adding content. Either access the replay directly, or by submitting and viewing from the teacher view.

### CODE REVIEW

File: apps/nextjs/src/components/scrubber.test.tsx [hunk 1]
```diff
@@ -3,6 +3,85 @@ import { describe, expect, it } from "vitest";
 import type { Range, TextDocumentPosition } from "./scrubber";
 import { deleteText, insertText } from "./scrubber";
 
+type ReplayEvent =
```
Comment (Krish120003) [inline apps/nextjs/src/components/scrubber.test.tsx]: Can't we use the types that @fvcci defined? @fvcci Can you help out here?

**Review by Krish120003 — APPROVED** [review]
